lib_collections/marklogic.py

```diff
@@ -0,0 +1,243 @@
+        Dictionary representing result of Mark Logic query, list of field
+        names (strings) stored in Wagtail database for a given collection
+
+    Returns:
+        Dictionary representing metadata for display in object page
+    """
+    filtered_names = [item
+                      for item in wagtail_field_names
+                      if item in dct.keys()]
+    updated_dict = {key: dct[key] for key in filtered_names}
+    return updated_dict
+
+
+def triples_to_dict(dct: dict,
+                    wagtail_field_names: list):
+    """
+    Capitalizes all metadata field names and sorts them using align_field_names
+    if the Wagtail database has information about which fields to display; if
+    not, displays all of them in alphabetical order.
+
+    Args: 
+        Dictionary represengint result of Mark Logic query, list of field
+        names (strings) stored in Wagtail database for a given collection
+
+    Returns: 
+        Reformatted dictionary representing metadata
+    """
+    try:
+        results = dct['results']['bindings'][0]
+        field_dict = {
+            key.capitalize(): results[key]['value'] for key in results.keys()
+        }
+        if wagtail_field_names:
+            return align_field_names(field_dict, wagtail_field_names)
+        else:
+            return field_dict
+    except KeyError:
+        raise Exception("Mark Logic result not formatted as expected")
+    except IndexError:
+        raise Http404
+
+
+def get_record_no_parsing(manifid: str,
+                          wagtail_field_names: list) -> dict:
+    """
+    Queries Mark Logic for collection object metadata fields, leaves the values
+    of those metadata fields alone, in their raw form.
+
+    Args: 
+        NOID string, list of field names
+
+    Returns:
+        Dictionary representing metadata
+    """
+    raw_record = get_raw_record(manifid)
+    return triples_to_dict(raw_record, wagtail_field_names)
+
+
+def remove_trailing_comma(string: str) -> str:
+    """
+    Remove leading/trailing whitespace and trailing commas from a string.
+
+    Args:
+        Metadata string
+
+    Returns:
+        Cleaned up metadata string
+    """
+    stripped = string.strip()
+    rev = stripped[::-1]
+    if len(rev) > 0 and rev[0] == ',':
+        return rev[1:][::-1]
+    else:
+        return rev[::-1]
+
+
+def brackets_parse(value: str) -> dict:
+    """
+    Parses the library cataloguers' notation for stating whether an author/date
+    attribution is externally derived.  Result is a representation of that
+    information in dictionary form.
+
+    Args:
+        Metadata string
+
+    Returns: 
+        Dictionary indicating whether author/date attributions are
+        externally derived
+    """
+    def question_mark(qm):
+        if qm is None:
+            return False
+        else:
+            return qm == '?'
+
+    prepped = remove_trailing_comma(value)
+    extract_from_brackets = re.compile("^\[([\w\s\d,.]*)(\\?)?\]")
```

Comment:
Thanks!  Sorry about that; I realized on just after putting all this up that my autofixers and linter had broken *yet again*.  After some sleuthing, I learned that I still had some residual `pip` installations of these apps that had gotten out of wack which my system-wide `python` installation, because my `emacs` is configured to run the OS-level versions of them rather than the virtualenv versions of them.  I think what I'm going to do short term is just make sure to install all these packages at the OS level rather than in pip.  Long-term, it would be nice to be able to make my editor aware of the virtualenv at the project level and run the versions of the autofixers installed within the project.  But I think the short-term solution will work for now.